Engine/Fuego/Windows/WindowWin.cpp

```diff
@@ -5,6 +5,66 @@
 
 namespace Fuego
 {
+
+DWORD __stdcall WindowWin::WinThreadMain(LPVOID lpParameter)
+{
+    WindowWin* _wnd = reinterpret_cast<WindowWin*>(lpParameter);
+
+    static TCHAR buffer[32] = TEXT("");
+#ifdef UNICODE
+    MultiByteToWideChar(CP_UTF8, 0, props.Title.c_str(), -1, buffer, _countof(buffer));
+#else
+    sprintf_s(buffer, _wnd->m_Props.Title.c_str());
+#endif
+    WNDCLASSEX wndClass = {};
+    wndClass.cbSize = sizeof(WNDCLASSEX);
+    wndClass.lpszClassName = _wnd->m_Props.APP_WINDOW_CLASS_NAME;
+    wndClass.hInstance = _wnd->m_HInstance;
+    wndClass.hIcon = LoadIcon(nullptr, IDI_WINLOGO);
+    wndClass.hCursor = LoadCursor(nullptr, IDC_ARROW);
+    wndClass.lpfnWndProc = WindowProcStatic;
+
+    RegisterClassEx(&wndClass);
+
+    DWORD style = WS_OVERLAPPEDWINDOW | WS_SYSMENU;
+    RECT rect;
+    rect.left = _wnd->m_Props.x;
+    rect.top = _wnd->m_Props.y;
+    rect.right = rect.left + _wnd->m_Props.Width;
+    rect.bottom = rect.top + _wnd->m_Props.Height;
+
+    AdjustWindowRect(&rect, style, true);
+
+    _wnd->m_Hwnd = CreateWindowEx(0, _wnd->m_Props.APP_WINDOW_CLASS_NAME, buffer, style, rect.left, rect.top, rect.right - rect.left, rect.bottom - rect.top,
+                                  nullptr, nullptr, _wnd->m_HInstance, nullptr);
+    FU_CORE_ASSERT(_wnd->m_Hwnd, "[AppWindow] hasn't been initialized!");
+    hwndMap.emplace(_wnd->m_Hwnd, _wnd);
+
+    _wnd->_context = new OpenGLContext(&_wnd->m_Hwnd);
```

Comment:
- Shouldn't use Opengl or any rendering API here. 
- Wait for thread to create window on main thread.